packages/comms/src/ecl/queryGraph.ts

```diff
@@ -0,0 +1,803 @@
+                    console.warn("Graph.getItem - Unknown Node Type!");
+                    break;
+            }
+        }
+        const retVal = this.idx[id];
+        Array.from(docNode.attributes).forEach(attr => {
+            retVal[attr.name] = attr.value;
+        });
+        return retVal;
+    }
+
+    removeItem(item: Subgraph | Vertex | Edge) {
+        delete this.idx[item.__hpcc_id];
+        if (item instanceof Subgraph) {
+            this.subgraphs = this.subgraphs.filter((subgraph: Subgraph) => {
+                return item !== subgraph;
+            });
+        } else if (item instanceof Vertex) {
+            this.vertices = this.vertices.filter(vertex => {
+                return item !== vertex;
+            });
+        } else if (item instanceof Edge) {
+            this.edges = this.edges.filter((edge: Edge) => {
+                return item !== edge;
+            });
+        }
+    }
+
+    getChildByTagName(docNode: HTMLElement, tagName: string): HTMLElement | null {
+        let retVal = null;
+        Array.from(docNode.childNodes).some((childNode: HTMLElement, idx) => {
+            if (childNode.tagName === tagName) {
+                retVal = childNode;
+                return true;
+            }
+        });
+        return retVal;
+    }
+
+    walkDocument(docNode: HTMLElement, id: string): Subgraph | Vertex | Edge {
+        const retVal: any = this.getItem(docNode, id);
+        docNode.childNodes.forEach((childNode: HTMLElement, idx) => {
+            switch (childNode.nodeType) {
+                case 1:     // 	ELEMENT_NODE
+                    switch (childNode.tagName) {
+                        case "graph":
+                            break;
+                        case "node":
+                            let isSubgraph = false;
+                            const attNode = this.getChildByTagName(childNode, "att");
+                            if (attNode) {
+                                const graphNode = this.getChildByTagName(attNode, "graph");
+                                if (graphNode) {
+                                    isSubgraph = true;
+                                    const subgraph = this.walkDocument(graphNode, childNode.getAttribute("id"));
+                                    retVal.addSubgraph(subgraph);
+                                }
+                            }
+                            if (!isSubgraph) {
+                                const vertex = this.walkDocument(childNode, childNode.getAttribute("id"));
+                                retVal.addVertex(vertex);
+                            }
+                            break;
+                        case "att":
+                            const name = childNode.getAttribute("name");
+                            const value = childNode.getAttribute("value");
+                            if (name.indexOf("Time") === 0) {
+                                retVal["_" + name] = value;
```

Comment:
Though it sometimes seems like they might not be actual issues, unless what its referring to can somehow be passed in via the url. Otherwise, you're just messing with your own browser session..?